internal/backend/db/dbgorm/gorm_test.go

```diff
@@ -21,10 +26,72 @@ import (
 	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
 )
 
-var now time.Time
+var (
+	now    time.Time
+	dbType string
+	gormDB gormdb
+)
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&dbType, "dbtype", "sqlite", "Database type to use for tests (e.g., sqlite, postgres)")
+	flag.Parse()
+
+	var pg *embedPG.EmbeddedPostgres
+
+	if dbType == "postgres" {
+		pg = embedPG.NewDatabase()
+		if err := pg.Start(); err != nil {
+			log.Fatalf("failed to start postgres: %v", err)
+		}
+		fmt.Println("Started PG..")
+	}
+
+	// setup test bench - gorm, schemas, migrations, etc
+	cfg := gormkitteh.Config{Type: dbType, DSN: ""} // "" for in-memory, or specify a file
+	db := setupDB(&cfg)
+	gormDB = gormdb{db: db, cfg: &cfg, mu: nil, z: zap.NewExample()}
+
+	ctx := context.Background()
+	if err := gormDB.Setup(ctx); err != nil { // ensure migration/schemas
+		log.Fatalf("Failed to setup gormdb: %v", err)
+	}
+
+	now = time.Now()
+	now = now.Truncate(time.Microsecond) // PG default resolution is microseconds
+
+	// PG saves dates in UTC. Gorm converts them back to local TZ on read
+	// SQLite has no dedicated time format and uses strings, so gorm will read them as UTC, thus we need to write them in UTC
+	if dbType == "sqlite" {
+		now = now.UTC()
+	}
+
+	// run tests
+	exitCode := m.Run()
+
+	// teardown test bench
```

Comment:
IIUC both databases are in memory, why do we need a teardown then?

Reply:
you could run tests on actual, persistent DBs as well. So I prefer normal cleanup process - create tables, run tests, destroy tables.